Fix LATIN-1 (iso-8859-1) encoding.

username1565 opened the pull request:
```
compare(new TextEncoder('iso-8859-1').encode(new TextDecoder('iso-8859-1').decode(allBytes)), allBytes); //can return true
```

username1565 committed:
How to fix here the LATIN-1 encoding (iso-8859-1)?

Test, my fix you can see here: https://username1565.github.io/text-encoding/lib/index.html

username1565 committed:
Latin-1 (iso-8859-1) encoding - fixed.

See the code changes in the two previous commits. This, and this: https://github.com/username1565/text-encoding/commit/510235dfd777a6a04d0360a871ee5df2e6f00e8a

odahcam commented:
Please, we do have some error with Latin1, please check [this test where our encoder fails](https://travis-ci.org/github/zxing-js/library/jobs/672574089) due to some Latin1 bug; it would be great if that gets fixed. We discussed the bug in here: https://github.com/zxing-js/library/pull/294

username1565 commented:
Delete this comment. I'll write another, correct comment, later.
_You can see edition history of this._

username1565 commented on lib/index.html:
```diff
@@ -0,0 +1,106 @@
+	function Latin1StrToUint8(Latin1Str){
+		var replaceCharCodesForLatin1 = {
+			//'character'	:		iso-8859-1 charcode,	//Unicode charcode(commented)
+				'€'			:		128,					//8364, 
```
`ISO-8859-1 (LATIN-1)`  not contains this symbol `€`. Maybe [Windows-1252 encoding](https://en.wikipedia.org/wiki/Windows-1252) used in TextEncoder, instead [`ISO-8859-1` (Latin-1)](https://en.wikipedia.org/wiki/ISO/IEC_8859-1#Code_page_layout).
See https://github.com/sinonjs/text-encoding/pull/2

P.S.:
Yeap, `iso-8859-1 (Latin-1)` this is interpretting, as `windows-1252`, here:
https://github.com/username1565/text-encoding/blob/ee7bc2de1c4589cbe0f8556ca1864119c435b7f5/lib/encoding.js#L600

Fix:
```
        {
          "labels": [
            "ansi_x3.4-1968",
            "cp1252",
            "cp819",
            "ibm819",
            "iso-ir-100",
            "windows-1252",
            "x-cp1252"
          ],
          "name": "windows-1252"
        },
        {
          "labels": [
            "ascii",
            "us-ascii",
            "iso-8859-1",
            "iso8859-1",
            "iso88591",
            "iso_8859-1",
            "iso_8859-1:1987",
            "l1",
            "latin1",
            "csisolatin1",
          ],
          "name": "iso-8859-1"
        },
```
and add in `encoding-indexes.js` the following string:
```
	//char codes of characters, after ASCII-chars:
//...
  "iso-8859-1":[128,129,130,131,132,133,134,135,136,137,138,139,140,141,142,143,144,145,146,147,148,149,150,151,152,153,154,155,156,157,158,159,160,161,162,163,164,165,166,167,168,169,170,171,172,173,174,175,176,177,178,179,180,181,182,183,184,185,186,187,188,189,190,191,192,193,194,195,196,197,198,199,200,201,202,203,204,205,206,207,208,209,210,211,212,213,214,215,216,217,218,219,220,221,222,223,224,225,226,227,228,229,230,231,232,233,234,235,236,237,238,239,240,241,242,243,244,245,246,247,248,249,250,251,252,253,254,255], 
```

username1565 commented on lib/index.html:
```diff
@@ -0,0 +1,106 @@
+
+	function Latin1StrToUint8(Latin1Str){
+		var replaceCharCodesForLatin1 = {
+			//'character'	:		iso-8859-1 charcode,	//Unicode charcode(commented)
```
This is `windows-1252` characters, not `iso-8859-1 (Latin-1)`.

username1565 committed:
Finally fix "iso-8859-1" encoding (Latin-1)

Finally fix "iso-8859-1" encoding (Latin-1):
encoding-indexes.js:
	Line 10:	Add comment about the following values - this is a charcodes of chars, after ASCII-chars.
	Line 18:	Add "iso-8859-1" with additional 128 consecutive chars, after ASCII-chars.
	
encoding.js:
	Lines 580	-	606		:	Using "iso-8859-1"-record from "encoding-indexes.js", instead "windows-1252", for "latin-1", and "ASCII".
	Lines 1245	-	1247	:	Add "iso-8859-1" as parameter for TextEncoder.
		Now, var bytes = new TextEncoder("iso-8859-1").encode(latin1_string); //not returning the throw error, and working.
	Lines 3305	-	3308	:	Override the browser's TextEncoder and TextDecoder, when this was been already defined in browser.
	
index.html:
	First script:
		Add string with latin-1 alphabet, usage to encoding and decoding chars, using this, and test (this all was been commented).
		Add characters table of latin-1 encoding, as object, and two functions to encode and decode strings, by using this:
			Latin1ToUint8(Latin1){		//encode latin-1 string to bytes
			Uint8ToLatin1(Uint8Arr){	//decode bytes to latin-1 string
		Add tests for this all.
	This functions can be used to encode and decode latin-1, when TextEncoder and TextDecoder not working good.
	
	Second script:
		Test Latin1-compatibility of TextEncoder, and TextDecoder, and fix "false", by using the previous functions.
	
That's all!

username1565 reviewed (commented):
Done.

odahcam commented on lib/encoding.js:
```diff
@@ -3297,6 +3302,10 @@
 
+  if (typeof TextEncoder === 'function') //override browser's TextEncoder, by this TextEncoder.
+    global['TextEncoder'] = TextEncoder;
+  if (typeof TextDecoder === 'function') //override browser's TextDecoder, by this TextDecoder.
```
What if I intend to use this as a module in an isolated scope? This will override the global TextEncoder/Decoder and may cause unexpected results. To avoid unexpected behaviors, I think this should be optional.

username1565 replied:
Yes!
For example, after override, [after all tests](https://username1565.github.io/text-encoding/lib/index.html), the command in console.log:
```
console.log(new TextEncoder('unicode').encode(decoded));
```
not working as before override, because 'unicode' not specified in `encoding-indexes.js`

This override used
just because this page: https://username1565.github.io/text-encoding/lib/index.html ([src](https://github.com/username1565/text-encoding/lib/index.html)) ,
show [(test === false), in console.log](https://github.com/username1565/text-encoding/blob/c1193bd1cc3fe52f826256a6b4985869526c50b9/lib/index.html#L217),
when `TextEncoder` and `TextDecoder` already defined in the browser.

Anyway, you can see my commits, make `fork` of this [from me](https://github.com/username1565/text-encoding),
then `edit` the source code of it as you want,
and `use` this `as module` from `your own` repositary.

odahcam commented:
When I commented https://github.com/sinonjs/text-encoding/pull/2#issuecomment-625555876 I really thought @sinonjs would take a look on the PR. Unfortunately it looks like it won't happen.

username1565 commented:
I see some activity in https://github.com/sinonjs/fake-timers/tree/master/integration-test
last month updated, and some @sinonjs's - repositaries here https://github.com/sinonjs was been updated few days ago.

odahcam committed:
add .gitignore

odahcam committed:
add puppeteer test runner

odahcam committed:
add travis runner

odahcam committed:
Rename travis.yml to .travis.yml

odahcam committed:
package scripts + code styles

odahcam committed:
Merge branch 'master' of https://github.com/zxing-js/text-encoding

odahcam committed:
node.js unit tests

odahcam committed:
fix assert param order

odahcam committed:
moved assert-package

odahcam committed:
fix editorconfig

odahcam committed:
drop python server dependency

odahcam committed:
background test server for CI

odahcam committed:
WIP initial port

pushing as a backup as I phisically am unable to finish this port, I'll try to finish this later

odahcam committed:
fix es2015, esm, cjs build so I can sleep in peace

odahcam committed:
restored script wiped out before commit

odahcam committed:
fixed build

odahcam committed:
fixed assert package path

odahcam committed:
Make test-server.sh executable

odahcam committed:
up travis ci

username1565 committed:
Merge pull request #2 from zxing-js/master

Automatically run tests on commit

username1565 committed:
Revert "Automatically run tests on commit"

username1565 committed:
Merge pull request #3 from username1565/revert-2-master

Revert "Automatically run tests on commit"

username1565 committed:
Revert "Revert "Automatically run tests on commit""

odahcam committed:
encodingIndexes

odahcam committed:
refactor encoding indexes accessing

odahcam committed:
crrected entrypoint

odahcam committed:
fixed build warns + full umd build

odahcam committed:
removed old plain js lib

odahcam committed:
moved node tests inside main folder

odahcam committed:
using new typescript compiled lib

odahcam committed:
removed __tests__

odahcam committed:
build library before testing

odahcam committed:
updated examples

odahcam committed:
updated package owner

odahcam committed:
removed old node entrypoint

odahcam committed:
entrypoints for node and browser

odahcam committed:
up ignore

odahcam committed:
moved browser tests

odahcam committed:
upd lib location + added indexes

odahcam committed:
removed old integration folder

odahcam committed:
global scope/variable access fix

username1565 committed:
test

test

username1565 committed:
Update .travis.yml

username1565 committed:
Update .travis.yml

username1565 committed:
Update test-server.sh

username1565 committed:
Update package.json

username1565 committed:
Update package.json

username1565 committed:
Update .travis.yml

username1565 committed:
Turn back this string

Turn back this string.

username1565 committed:
Merge pull request #4 from username1565/revert-3-revert-2-master

Add travis-ci tests for each commit

username1565 committed:
Update README.md

username1565 committed:
Update README.md

username1565 committed:
Update README.md

username1565 committed:
Fix build status-imae for "master"-branch

Fix build status-imae for "master"-branch

username1565 committed:
Upload original file, to discard all changes.

Upload original file, to discard all changes.

username1565 committed:
Upload original file, to discard all changes.

Upload original file, to discard all changes.
Because images associated with my fork and branch,
and can return invalid result for another forks,
and can be changed, after accepting the Pull Requests.

odahcam committed:
fix jest ArrayBuffer bug

odahcam committed:
fix encodingIndex global access

odahcam committed:
proper named encoding-indexes script

odahcam committed:
fix instantiation and action methods

odahcam committed:
added jest debug launcher

odahcam committed:
misc

odahcam committed:
improved tests startup

odahcam committed:
pkg and build adjustments

odahcam committed:
changed main module source to cjs

odahcam committed:
fixed throw assertions

odahcam committed:
fixed encoding-factory main exports

odahcam committed:
not creating encoding-indexes source map for umd

odahcam committed:
proper array buffer types test

odahcam committed:
polyfills

odahcam committed:
not used code

odahcam committed:
improved array equals

odahcam committed:
setup closest to original package

odahcam committed:
skipping erroring test for now

odahcam committed:
Create FUNDING.yml

odahcam committed:
added typescript

odahcam committed:
Merge branch 'feature/typescript'

username1565 committed:
Update .travis.yml

username1565 committed:
Update .travis.yml

username1565 committed:
Update .travis.yml

username1565 committed:
Update .travis.yml

username1565 committed:
Update .travis.yml

username1565 committed:
Merge pull request #7 from zxing-js/master

Update "master"-branch

username1565 committed:
Revert "Update "master"-branch"

username1565 committed:
Merge pull request #8 from username1565/revert-7-master

Revert "Update "master"-branch"

username1565 committed:
Update .travis.yml

username1565 committed:
Update .travis.yml

username1565 committed:
Change pathway to '../util/assert-package.js'

Change pathway from '../assert-package.js' to '../util/assert-package.js'

username1565 committed:
Add testharness.js

Add testharness.js

username1565 committed:
Delete LICENSE

username1565 committed:
Delete idlharness.js

username1565 committed:
Delete readme.md

mroderick commented:
> When I commented [#2 (comment)](https://github.com/sinonjs/text-encoding/pull/2#issuecomment-625555876) I really thought @sinonjs would take a look on the PR. Unfortunately it looks like it won't happen.

We've een a bit distracted lately, as I am sure you can understand.

username1565 commented:
I just close this pull request, because I did delete my old repositary.
@odahcam can open new Pull Request for @sinonjs
